DQM/L1TMonitor/src/L1TObjectsTiming.cc

```diff
@@ -0,0 +1,483 @@
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    tau_eta_phi_lastbunch.push_back(ibooker.book2D("tau_eta_phi_bx_lastbunch_"+bx_obj[i+2],"L1T Tau #eta vs #phi for last bunch BX="+bx_obj[i+2],25, -2.5, 2.5, 25, -3.2, 3.2));
+    tau_eta_phi_lastbunch.at(i)->setAxisTitle("#eta", 1);
+    tau_eta_phi_lastbunch.at(i)->setAxisTitle("#phi", 2);
+  }
+
+  ibooker.setCurrentFolder(monitorDir+"/L1TEtSum"+"/timing"+"/Last_bunch");
+  for(unsigned int i=0; i<bxrange-2; ++i) {
+    etsum_eta_phi_MET_lastbunch.push_back(ibooker.book1D("etsum_phi_bx_MET_lastbunch_"+bx_obj[i+2],"L1T MET #phi for lastbunch bunch BX="+bx_obj[i+2],25, -3.2, 3.2));
+    etsum_eta_phi_MET_lastbunch.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_METHF_lastbunch.push_back(ibooker.book1D("etsum_phi_bx_METHF_lastbunch_"+bx_obj[i+2],"L1T METHF #phi for lastbunch bunch BX="+bx_obj[i+2],25, -3.2, 3.2));
+    etsum_eta_phi_METHF_lastbunch.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_MHT_lastbunch.push_back(ibooker.book1D("etsum_phi_bx_MHT_lastbunch_"+bx_obj[i+2],"L1T MHT #phi for lastbunch bunch BX="+bx_obj[i+2],25, -3.2, 3.2));
+    etsum_eta_phi_MHT_lastbunch.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_MHTHF_lastbunch.push_back(ibooker.book1D("etsum_phi_bx_MHTHF_lastbunch_"+bx_obj[i+2],"L1T MHTHF #phi for lastbunch bunch BX="+bx_obj[i+2],25, -3.2, 3.2));
+    etsum_eta_phi_MHTHF_lastbunch.at(i)->setAxisTitle("#phi", 1);
+  }
+} 
+
+}
+
+  
+
+void L1TObjectsTiming::analyze(const edm::Event& e, const edm::EventSetup& c) {
+
+  if (verbose) edm::LogInfo("L1TObjectsTiming") << "L1TObjectsTiming: analyze..." << std::endl;
+
+
+  // Muon Collection
+  edm::Handle<l1t::MuonBxCollection> MuonBxCollection;
+  e.getByToken(ugmtMuonToken_, MuonBxCollection);
+  // Jet Collection
+  edm::Handle<l1t::JetBxCollection> JetBxCollection;
+  e.getByToken(stage2CaloLayer2JetToken_, JetBxCollection);
+  // EGamma Collection
+  edm::Handle<l1t::EGammaBxCollection> EGammaBxCollection;
+  e.getByToken(stage2CaloLayer2EGammaToken_, EGammaBxCollection);
+  // Tau Collection
+  edm::Handle<l1t::TauBxCollection> TauBxCollection;
+  e.getByToken(stage2CaloLayer2TauToken_, TauBxCollection);
+  // EtSum Collection
+  edm::Handle<l1t::EtSumBxCollection> EtSumBxCollection;
+  e.getByToken(stage2CaloLayer2EtSumToken_, EtSumBxCollection);
+
+  // Open uGT readout record
+  edm::Handle<GlobalAlgBlkBxCollection> uGtAlgs;
+  e.getByToken(l1tStage2uGtProducer_, uGtAlgs);
+          
+
+    // Filling eta-phi map for muons for BX=-2,-1,0,+1,+2
+    for (int itBX = MuonBxCollection->getFirstBX(); itBX <= MuonBxCollection->getLastBX(); ++itBX) {
+      for (l1t::MuonBxCollection::const_iterator Muon = MuonBxCollection->begin(itBX); Muon != MuonBxCollection->end(itBX); ++Muon) {
+        int index = (int)itBX - std::min(0, 1 - (int)bxrange%2 - (int)std::floor(bxrange/2.)); // the correlation from itBX to respective index of the vector
+        muons_eta_phi.at(index)->Fill(Muon->eta(), Muon->phi());
```

Comment:
You could move the filling of the isolated, first, and last bunch histograms also in here.
```
    for(GlobalAlgBlkBxCollection::const_iterator itr = uGtAlgs->begin(0); itr != uGtAlgs->end(0); ++itr) {
      if((algoBitFirstBxInTrain_ != -1 && algoBitLastBxInTrain_ != -1) && (itr->getAlgoDecisionInitial(algoBitFirstBxInTrain_) && itr->getAlgoDecisionInitial(algoBitLastBxInTrain_))) {
        muons_eta_phi_isolated.at(index)->Fill(muon->eta(),muon->phi());
      }
      if (algoBitFirstBxInTrain_ != -1 && itr->getAlgoDecisionInitial(algoBitFirstBxInTrain_) && itBX <= 0) {
        muons_eta_phi_firstbunch.at(index)->Fill(muon->eta(),muon->phi());
      }
      if(algoBitLastBxInTrain_ != -1 && itr->getAlgoDecisionInitial(algoBitLastBxInTrain_) && itBX >= 0) {
        muons_eta_phi_lastbunch.at(index)->Fill(muon->eta(),muon->phi());
      }
    }
```
The same works for all other L1T objects. This would reduce the lines of code.